src/components/EventTheme/colors.tsx

```diff
@@ -0,0 +1,67 @@
+export default {
+  charlotte: {
+    "color-backdrop-subtle": "#72be8d",
+    "color-faint": "#abd9ba",
+    "color-primary": "#1e7640",
+    "color-primary-dark": "#154427",
+    "color-primary-light": "#afe9c4",
+  },
+  london: {
+    "color-backdrop-subtle": "#2a45a3",
+    "color-faint": "#ebf0f7",
+    "color-primary": "#22366e",
+    "color-primary-dark": "#19223a",
+    "color-primary-light": "#7083c6",
+  },
+  newquay: {
+    "color-backdrop-subtle": "#262262",
+    "color-faint": "#e4e4ec",
+    "color-primary": "#262261",
+    "color-primary-dark": "#252347",
+    "color-primary-light": "#5a5880",
+  },
+  newyork: {
+    "color-backdrop-subtle": "#2f2d2e",
+    "color-faint": "#cbcbcb",
+    "color-primary": "#2f2d2e",
+    "color-primary-dark": "#151414",
+    "color-primary-light": "#5b5457",
+  },
+  phoenix: {
+    "color-backdrop-subtle": "#524ca6",
+    "color-faint": "#e4e4ec",
+    "color-primary": "#262262",
+    "color-primary-dark": "#252347",
+    "color-primary-light": "#736ccd",
+  },
+  telaviv: {
+    "color-backdrop-subtle": "#5073df",
+    "color-faint": "#9cb0d9",
+    "color-primary": "#324d9c",
+    "color-primary-dark": "#28396a",
+    "color-primary-light": "#809bed",
+  },
+  vienna: {
+    "color-backdrop-subtle": "#e78689",
+    "color-faint": "#fadddf",
+    "color-primary": "#b41c21",
+    "color-primary-dark": "#8b0f13",
+    "color-primary-light": "#f3a7a9",
+  },
+  belgrade: {
+    "color-backdrop-subtle": "#e1ff99",
+    "color-faint": "#e1ff99",
+    "color-primary": "#7aa11c",
+    "color-primary-dark": "#304007",
+    "color-primary-light": "#e1ff99",
+  },
+} as Record<
```

Comment:
[Suggestion] Heh, the `as` type assertion is because keys of type `string` can't be used to index into this object, right? Until #18 is in I suppose there will always need to be some assertions. A fewer-lines-of-code solution would be to, instead of `as` asserting here, using an `as` in `EventColors`:

```ts
return colors[event as keyof typeof colors];
```

That way also has the added benefit of letting `colors` keep its more specific type. Which might be useful later on.


<details>
<summary>If we really wanted to, we could go through the effort of making it more safe...</summary>
I think TypeScript is right to yell here: what happens if someone makes a new event -say, `"philadelphia"`- and forgets to add to `colors.tsx`? The page will crash. I simulated this by deleting the `charlotte` entry and visiting `localhost:3000/charlotte` locally:

```plaintext
TypeError: Cannot read properties of undefined (reading 'color-backdrop-subtle')

src/pages/[event]/index.tsx (124:34) @ Event
  122 |     geolocation={geolocation}
  123 |     slug={slug}
> 124 |     waterColor={eventColors["color-backdrop-subtle"]}
      |                            ^
  125 |   />
  126 | )}
```

An alternative might be to instead change from `event: string` to a more specific type, like `event: EventName` or `event: EventSlug` or some name like that.

One way you could do that would be to:
1. Change `src/data/events.json` to `src/data/events.ts`
2. Have it `export const events = [...] as const` _(maybe even renaming it to `eventsOrder`, since that's what it's used for?)_
3. Also have `events.ts` `export const EventName = typeof events[number]`
4. Use that `EventName` type wherever something is meant to be an event

Note that this includes `getEvents()` in `src/data/index.ts`, which would need to `as EventSlug[]`... which I guess means there's always going to be be a type assertion until #18 is in.
</details>

Reply:
Love this, going to give it a try. I think it'll end up being a bit bigger PR though.

Learned a lot of TS on this change so far. Didn't know you could do this with `as const`.